docs/content/concepts/assets/asset-checks/define-execute-asset-checks.mdx

```diff
@@ -0,0 +1,331 @@
+---
+title: "Defining and executing asset checks | Dagster Docs"
+description: Asset checks are a way to define expectations about the quality of your data.
+---
+
+# Defining and executing asset checks (Experimental)
+
+<ExperimentalCallout />
+
+After creating some [Software-defined Assets](/concepts/assets/software-defined-assets), you may want to automate checks on the assets that test for data quality.
+
+In this guide, we'll show you a few approaches to defining asset checks, how to use check results to include helpful information, and how to execute checks.
+
+---
+
+## Defining asset checks
+
+There are a few methods you can use to define an asset check:
+
+- **Define separate functions for assets and checks** - In this approach, asset materialization and asset checks are executed in their own separate ops. If using the <PyObject object="multiprocess_executor" />, this allows you to launch runs that will use separate processes to materialize the asset and execute its check.
+- **Define assets and checks in the same function** - In this approach, checks execute in the same op that materializes the asset.
+- **Define an asset check factory** - Useful for defining multiple, similar asset checks at once
```

Comment:
I took a swing at making the titles a bit more specific/digestible (ex: original used to be "Checks that execute in their own ops", but I don't know what that means) but I'm not sure if these are correct or more helpful. Suggestions welcome.

Reply:
Agreed that the current titles are not very digestible. I always get hung up on this function vs. op thing. Thoughts on something like "Defining checks separately from the assets they target" and "Defining assets and checks together"?

Reply:
It is kinda weird - do users need to know about it being an op? Would a dev go about anything differently because they knew this, or can we just say _"Defining assets separately from the assets they target will do <things listed in the first bullet>"?_

Reply:
> do users need to know about it being an op?

I think many users can be successful without knowing this.

It's useful context when you're trying to understand what's occurring when you execute a run that both materializes assets and executes checks.  The run page shows a list of steps, and each op results in exactly 1 step in runs that include that op.

Can chat this through in more detail if helpful